### PR TITLE
Search other pages and take user to results selected from those pages 

### PR DESCRIPTION
**Pull request by Jaxobc**

### CODE REVIEW

File: portfolio/src/main/webapp/js/custom-script.js [hunk 1]
```diff
@@ -229,15 +227,65 @@ function allowUserSubmit() {
...
+ * Gets the other html pages as documents and invokes method to insert Search
+ * bar
+ */
+async function loadSearch() {
+  const otherDocs =
+      await Promise.all([getHTML('/about.html'), getHTML('/statistics.html')]);
+
+  let otherElements = [];
+  otherDocs.forEach((doc) => {
+    otherElements = [...otherElements, ...Array.from(doc.body.childNodes)];
```
Comment (briafassler9) [inline portfolio/src/main/webapp/js/custom-script.js:311]: Just out of my own curiosity, what is the functionality of the ellipsis here?

File: portfolio/src/main/webapp/js/custom-script.js [hunk 2]
```diff
@@ -229,15 +227,65 @@ function allowUserSubmit() {
...
+ * Get HTML asynchronously as document
+ * @param {String} url: The URL to get HTML from
+ */
+async function getHTML(url) {
+  return new Promise((resolve, reject) => {
+    // Feature detection
+    if (!window.XMLHttpRequest) return;
+
+    // Create new request
+    const xhr = new XMLHttpRequest();
```
Comment (briafassler9) [inline portfolio/src/main/webapp/js/custom-script.js:329]: What is xhr? A more specific variable name would be helpful

File: portfolio/src/main/webapp/js/custom-script.js [hunk 3]
```diff
@@ -229,15 +227,65 @@ function allowUserSubmit() {
...
+}
+
+/**
+ * Get HTML asynchronously as document
+ * @param {String} url: The URL to get HTML from
+ */
+async function getHTML(url) {
+  return new Promise((resolve, reject) => {
+    // Feature detection
+    if (!window.XMLHttpRequest) return;
```
Comment (briafassler9) [inline portfolio/src/main/webapp/js/custom-script.js]: nit: other if statements throughout the pr (ex. lines 368-371) are not single lined. just for consistency i'd change this if statement or the other if statements

File: portfolio/src/main/webapp/js/custom-script.js (hunk 1 above)
Comment (kofimeighan) [reply portfolio/src/main/webapp/js/custom-script.js:311]: +1

File: portfolio/src/main/webapp/js/custom-script.js [hunk 4]
```diff
@@ -229,15 +227,65 @@ function allowUserSubmit() {
...
 
-/* inserts a functioning searchbar into the navigation bar of a page. */
-function insertSearch() {
+/**
+ * Gets the other html pages as documents and invokes method to insert Search
+ * bar
+ */
+async function loadSearch() {
+  const otherDocs =
+      await Promise.all([getHTML('/about.html'), getHTML('/statistics.html')]);
```
Comment (kofimeighan) [inline portfolio/src/main/webapp/js/custom-script.js:307]: what about if you load this on the statistics page? how will it get the html from index.html?

File: portfolio/src/main/webapp/js/custom-script.js (hunk 3 above)
Comment (kofimeighan) [reply portfolio/src/main/webapp/js/custom-script.js]: +1, for consistency i'd prefer to add brackets and make it multi line

File: portfolio/src/main/webapp/js/custom-script.js [hunk 5]
```diff
@@ -303,21 +354,47 @@ function showResults(resultElements, wantedWords) {
...
+    const textElement = createResult(result, wantedWords)[1];
     textElement.onclick = function() {
       result.scrollIntoView();
     };
 
     searchResults.appendChild(textElement);
   });
 }
+
+function showOtherResults(otherResultElements, wantedWords) {
```
Comment (kennzoid) [inline portfolio/src/main/webapp/js/custom-script.js:453]: Add docstrings to these new functions. In particular I'd like to see the difference between showOtherResults and showResults described _somewhere_.

**Review by kennzoid — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/js/custom-script.js (hunk 1 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js:311]: It is just the format to combine arrays.

File: portfolio/src/main/webapp/js/custom-script.js (hunk 2 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js:329]: The request returns an XMLHttpRequest (XHR) object, thus its the most appropriate name.

File: portfolio/src/main/webapp/js/custom-script.js (hunk 4 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js:307]: My goal is to have it working for the index page before making it work for every page.

File: portfolio/src/main/webapp/js/custom-script.js (hunk 3 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js]: Okay sure, I can change that.

File: portfolio/src/main/webapp/js/custom-script.js (hunk 5 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js:453]: Okay sure, I'll add those in.

**Review by kofimeighan — APPROVED** [review]

**Review by briafassler9 — APPROVED** [review]

**Review by kennzoid — APPROVED** [review]

**Pull request merged**